### PR TITLE
Refactor: finishing the move to the sidecar pattern

### PR DESCRIPTION
**Pull request by jdkandersson**
- Update the metadata to be compliant with the sidecar pattern
- Add charmcraft.yaml to enable the charmcraft pack command
- Minor changes to the source code
- Updated .gitignore to remove venv and .vscode

### CODE REVIEW

**Comment by jdkandersson** [comment]
This repo is still in early days, e.g., no ci etc. Still worth doing PRs regularly to get feedback.

File: metadata.yaml [hunk 1]
```diff
@@ -17,4 +18,5 @@ containers:
 resources:
   jenkins-agent-image:
     type: oci-image
-    description: OCI image for jenkins-agent
\ No newline at end of file
+    description: OCI image for jenkins-agent
+    auto-fetch: true
```
Comment (mthaddon) [inline metadata.yaml:22]: What does this do? I don't see it mentioned on https://juju.is/docs/sdk/metadata-reference

File: metadata.yaml [hunk 2]
```diff
@@ -1,9 +1,10 @@
-name: alejdg-jenkins-agent-k8s
+name: jenkins-agent
```
Comment (mthaddon) [inline metadata.yaml:1]: The name should be `jenkins-agent-k8s` as this will replace https://charmhub.io/jenkins-agent-k8s.

**Review by mthaddon — COMMENTED** [review]
Some minor comments/questions

File: metadata.yaml (hunk 1 above)
Comment (jdkandersson) [reply metadata.yaml:22]: I'm not sure, I just copied the indico template file and adjusted it for jenkins agent: https://github.com/canonical/indico-operator/blob/improve-docs/metadata.yaml. I found it referenced here, it seems to somehow speed up docker interactions although I didn't quite follow the explanation: https://github.com/juju-solutions/layer-docker-resource

File: metadata.yaml (hunk 2 above)
Comment (jdkandersson) [reply metadata.yaml]: Thanks, fixed

**Review by mthaddon — APPROVED** [review]
LGTM

**Pull request merged**